solr/core/src/java/org/apache/solr/search/CaffeineCache.java

```diff
@@ -387,10 +388,16 @@ public void setMaxRamMB(int maxRamMB) {
     }
   }
 
-  protected void adjustMetrics(long hitsAdjust, long insertsAdjust, long lookupsAdjust) {
+  public void adjustMetrics(
+      long hitsAdjust, long insertsAdjust, long lookupsAdjust, CacheStats stats) {
     hits.add(-hitsAdjust);
     inserts.add(-insertsAdjust);
     lookups.add(-lookupsAdjust);
+    if (stats != null) {
+      synchronized (offsetSyncStats) {
```

Comment:
I assume the array of one element is the most efficient way to lock on a wrapped instance?

It always throws me off a little bit with array of 1 element, Im wondering what do u think about 
`AtomticRef` with `updateAndGet`? 😁

Reply:
Love it. Thanks for the suggestion!